packages/codemod/src/transforms/react-default-props-migration.ts

```diff
@@ -0,0 +1,426 @@
+          // INLINE -- MyComponent.defaultProps: { prop1: 'value1', prop2: 'value2' }
+          // Extract default props keys/values
+          defaultPropsMap = extractDefaultPropsFromNode(defaultPropsNode);
+        } else {
+          console.warn(
+            `[WARNING]: Do not know how to process default props for component "${componentName}": ${j(
+              path
+            ).toSource()}`
+          );
+        }
+
+        // 3. Next we update the component function signature
+        // We first look for classic function declarations
+        //   function MyComnponent(props) { ... }
+        const functionComponentDeclaration = root.find(j.FunctionDeclaration, {
+          id: { name: componentName },
+        });
+        if (functionComponentDeclaration.length === 1) {
+          functionComponentDeclaration.nodes()[0].params[0] =
+            transformComponentFunctionSignature({
+              functionPropsParam:
+                functionComponentDeclaration.nodes()[0].params[0],
+              defaultPropsMap,
+              componentName,
+              j,
+            });
+
+          // Get the function body scope so we only do the replacement
+          // within the component function we're currently processing
+          functionScope = j(functionComponentDeclaration.nodes()[0].body);
+        } else {
+          // If we don't find a function declaration, we look for arrow function declarations
+          //   const MyComnponent = (props) => { ... }
+          //   const MyComnponent = ({ prop1, ...props }) => { ... }
+          const variableComponentDeclaration = root.find(
+            j.VariableDeclaration,
+            {
+              declarations: [
+                {
+                  id: { name: componentName },
+                },
+              ],
+            }
+          );
+          if (variableComponentDeclaration.length === 1) {
+            const functionFirstParamNode =
+              variableComponentDeclaration.nodes()[0].declarations[0];
+
+            if (
+              functionFirstParamNode.type === 'VariableDeclarator' &&
+              functionFirstParamNode.init?.type === 'ArrowFunctionExpression'
+            ) {
+              functionFirstParamNode.init.params[0] =
+                transformComponentFunctionSignature({
+                  functionPropsParam: functionFirstParamNode.init.params[0],
+                  defaultPropsMap,
+                  componentName,
+                  j,
+                });
+
+              // Get the function body scope so we only do the replacement
+              // within the component function we're currently processing
+              functionScope = j(functionFirstParamNode.init.body);
+            } else {
+              console.warn(
+                `[WARNING]: Could parse component function first parameter "${componentName}"`
+              );
+            }
+          }
+        }
```

Comment:
I can follow along. Took me a while to realize - thanks to the comments - that we need to handle different ways in which React.FCs can be written.